src/FastRouteRouter.php

```diff
@@ -80,15 +109,31 @@ class FastRouteRouter implements RouterInterface
      *     implementation will be used.
      * @param null|callable $dispatcherFactory Callable that will return a
      *     FastRoute dispatcher.
+     * @param array $config Array of custom configuration options.
      */
-    public function __construct(RouteCollector $router = null, callable $dispatcherFactory = null)
-    {
+    public function __construct(
+        RouteCollector $router = null,
+        callable $dispatcherFactory = null,
+        array $config = null
+    ) {
         if (null === $router) {
             $router = $this->createRouter();
         }
 
         $this->router = $router;
         $this->dispatcherCallback = $dispatcherFactory;
+
+        if (isset($config['cache_enabled'])) {
+            $this->cacheEnabled = (bool) $config['cache_enabled'];
+        }
+
+        if (isset($config['cache_file'])) {
```

Comment:
> Why class constants?

Preventing typos, reducing documentation overhead, reducing hidden knowledge, easing refactoring

 > I meant those keys to be used (and related values loaded from) in configuration files, same way it's done for instance in https://github.com/zendframework/zend-expressive-zendviewrenderer/blob/master/src/HelperPluginManagerFactory.php.

If I had reviewed [`HelperPluginManagerFactory.php`](https://github.com/zendframework/zend-expressive-zendviewrenderer/blob/ca34ef052cefb48de5de6c519e61bbaf7870ea97/src/HelperPluginManagerFactory.php) when it was written, I would have raised the same issue during review

 > then I'd have to use it in config files too

You can still use string keys in config files, but it would be silly and error-prone.